src/middlewared/middlewared/plugins/pool.py

```diff
@@ -1406,23 +1457,15 @@ async def import_pool(self, job, data):
         else:
             pool_name = new_name
 
-        # reset top-level zpool `aclinherit` property
-        opts = {'properties': {'aclinherit': {'value': 'passthrough'}}}
-        await self.middleware.call('zfs.dataset.update', pool_name, opts)
+        ds = await self.middleware.call(
+            'pool.dataset.query',
+            [('id', '=', pool_name)],
+            {'get': True, 'extra': {'retrieve_children': False}}
+        )
+        ds_defaults_version = int(ds.get('truenas_defaults_version', {'value': '0'})['value'])
 
-        # Recursively reset dataset mountpoints for the zpool.
-        recursive = True
-        for child in await self.middleware.call('zfs.dataset.child_dataset_names', pool_name):
-            if child == os.path.join(pool_name, 'ix-applications'):
-                # We exclude `ix-applications` dataset since resetting it will
-                # cause PVC's to not mount because "mountpoint=legacy" is expected.
-                continue
-            try:
-                # Reset all mountpoints
-                await self.middleware.call('zfs.dataset.inherit', child, 'mountpoint', recursive)
-            except Exception as e:
-                # Let's not make this fatal
-                self.logger.warning('Failed to inherit mountpoints recursively for %r dataset: %r', child, e)
+        if ds_defaults_version != TRUENAS_DEFAULTS_VERSION:
```

Comment:
This check will prevent us from resetting bad properties set via CLI after the pool first import.

Reply:
I thought primary concern was users importing foreign pools?

Reply:
Foot-shooters are also a big concern, aren't they?